From 38b2ad72c1926c45790e678ce8c6207b3a93b2e5 Mon Sep 17 00:00:00 2001
From: Frantisek Sumsal <frantisek@sumsal.cz>
Date: Tue, 13 Jan 2026 17:45:45 +0100
Subject: [PATCH] Revert "coredump: verify pidfd after parsing data in usermode
 helper"

This reverts commit ae1394abcded51ed5e443d1124059b2e31748baa.

Reverts: RHEL-104135
---
 src/coredump/coredump.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/coredump/coredump.c b/src/coredump/coredump.c
index db7f76f6c4..d3a1f7c09d 100644
--- a/src/coredump/coredump.c
+++ b/src/coredump/coredump.c
@@ -1458,11 +1458,6 @@ static int gather_pid_metadata_from_procfs(struct iovec_wrapper *iovw, Context *
         if (get_process_environ(pid, &t) >= 0)
                 (void) iovw_put_string_field_free(iovw, "COREDUMP_ENVIRON=", t);
 
-        /* Now that we have parsed info from /proc/ ensure the pidfd is still valid before continuing */
-        r = pidref_verify(&context->pidref);
-        if (r < 0)
-                return log_error_errno(r, "PIDFD validation failed: %m");
-
         /* we successfully acquired all metadata */
         return context_parse_iovw(context, iovw);
 }
